verification/cbmc/sources/make_common_data_structures.c

```diff
@@ -33,9 +33,15 @@ void ensure_ring_buffer_has_allocated_members(struct aws_ring_buffer *ring_buf,
     size_t position_tail;
     __CPROVER_assume(position_head < size);
     __CPROVER_assume(position_tail < size);
-    aws_atomic_store_ptr(&ring_buf->head, (ring_buf->allocation + position_head));
-    aws_atomic_store_ptr(&ring_buf->tail, (ring_buf->allocation + position_tail));
-    ring_buf->allocation_end = ring_buf->allocation + size;
+    if (ring_buf->allocation) {
```

Comment:
```c
if (ring_buf->allocation != NULL) {
```

Reply:
Just curious: will it be more efficient or just to be more clear?

Reply:
This is nitpic, just for clarity. We should follow the pattern set by the maintainers.